lib/components/viewers/pattern-viewer.tsx

```diff
@@ -67,6 +67,23 @@ const PatternViewer = ({
   // @ts-expect-error TODO: add type to ComponentContext
   const { ModeIcon, RouteRenderer } = useContext(ComponentContext)
 
+  const routePatternKeys = route?.patterns && Object.keys(route?.patterns)
+
+  // If the patternId does not exist in the route, course correct back to a valid pattern.
+  // (ex. the URL was /route/123/undefined)
+  if (
+    viewedRoute?.patternId &&
+    !routePatternKeys?.includes(viewedRoute?.patternId) &&
+    routePatternKeys &&
+    routePatternKeys.length > 0
```

Comment:
These two checks are redundant with line 76.

Reply:
Taking away the check for `routePatternKeys` is still throwing a typescript error but I got rid of line 78! thanks!